streaming/base/stream.py

```diff
@@ -145,6 +149,15 @@ def __init__(self,
             self.keep_zip = keep_zip
             self.safe_keep_zip = self.keep_zip or self.remote in {None, self.local}
 
+    def _get_temporary_directory(self) -> str:
+        root = tempfile.gettempdir()
+        hash = None
+        if self.remote is not None:
+            hashlib.md5(self.remote.encode('utf-8')).hexdigest()
```

Comment:
What is the max character length output of `hashlib.md5` ?

Reply:
32